packages/core/src/external/commonwell/document/document-downloader-local.ts

```diff
@@ -0,0 +1,248 @@
+import { CommonWellAPI, CommonwellError, organizationQueryMeta } from "@metriport/commonwell-sdk";
+import AWS from "aws-sdk";
+import * as stream from "stream";
+import { DOMParser } from "xmldom";
+import { MetriportError } from "../../../util/error/metriport-error";
+import { makeS3Client, S3Utils } from "../../aws/s3";
+import {
+  Document,
+  DocumentDownloader,
+  DocumentDownloaderConfig,
+  DownloadResult,
+  FileInfo,
+} from "./document-downloader";
+
+export type DocumentDownloaderLocalConfig = DocumentDownloaderConfig & {
+  commonWell: {
+    api: CommonWellAPI;
+    queryMeta: ReturnType<typeof organizationQueryMeta>;
+  };
+};
+
+export class DocumentDownloaderLocal extends DocumentDownloader {
+  readonly s3client: AWS.S3;
+  readonly s3Utils: S3Utils;
+  readonly cwApi: CommonWellAPI;
+  readonly cwQueryMeta: ReturnType<typeof organizationQueryMeta>;
+
+  constructor(config: DocumentDownloaderLocalConfig) {
+    super(config);
+    this.s3client = makeS3Client(config.region);
+    this.cwApi = config.commonWell.api;
+    this.cwQueryMeta = config.commonWell.queryMeta;
+    this.s3Utils = new S3Utils(config.region);
+  }
+
+  override async download({
+    document,
+    fileInfo,
+  }: {
+    document: Document;
+    fileInfo: FileInfo;
+  }): Promise<DownloadResult> {
+    let downloadedDocument = "";
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const onData = (chunk: any) => {
+      downloadedDocument += chunk;
+    };
+    const onEnd = () => {
+      console.log("Finished downloading document");
+    };
+    const downloadResult = await this.downloadFromCommonwellIntoS3(
+      document,
+      fileInfo,
+      onData,
+      onEnd
+    );
+
+    const newlyDownloadedFile: DownloadResult = {
+      bucket: downloadResult.bucket,
+      key: downloadResult.key,
+      location: downloadResult.location,
+      size: downloadResult.size,
+      contentType: downloadResult.contentType,
+    };
+
+    if (
+      downloadedDocument &&
+      (document.mimeType === "application/xml" || document.mimeType === "text/xml")
```

Comment:
nit: could put the check into a function in our `core` package like:

```
mimeTypeIsXML(mimeType) {
   return mimeType === "application/xml" || mimeType === "text/xml");
}
```

this way we will never forget to check for both mimetypes, I remember we had an issue recently where we forgot to check for `text/xml` and it broke some logic


also, we could but `"text/xml"` and etc into the core package, and re-use those string consts (I see 2 other places where it's used in this PR)